src/EFCore.PG/Metadata/PostgresEnum.cs

```diff
@@ -9,102 +9,126 @@
         {
-            _annotatable = annotatable;
-            _annotationName = annotationName;
+            _annotatable = Check.NotNull(annotatable, nameof(annotatable));
+            _annotationName = Check.NotNull(annotationName, nameof(annotationName));
         }
 
+        /// <summary>
+        /// Gets or adds a <see cref="PostgresEnum"/> from or to the <see cref="IMutableAnnotatable"/>.
+        /// </summary>
+        /// <param name="annotatable">The annotatable from which to get or add the enum.</param>
+        /// <param name="schema">The enum schema or null to use the model's default schema.</param>
+        /// <param name="name">The enum name.</param>
+        /// <param name="labels">The enum labels.</param>
+        /// <returns>
+        /// The <see cref="PostgresEnum"/> from the <see cref="IMutableAnnotatable"/>.
+        /// </returns>
+        /// <exception cref="ArgumentException"><paramref name="schema"/></exception>
+        /// <exception cref="ArgumentNullException"><paramref name="annotatable"/></exception>
+        /// <exception cref="ArgumentNullException"><paramref name="name"/></exception>
+        /// <exception cref="ArgumentNullException"><paramref name="labels"/></exception>
+        [NotNull]
         public static PostgresEnum GetOrAddPostgresEnum(
             [NotNull] IMutableAnnotatable annotatable,
             [CanBeNull] string schema,
             [NotNull] string name,
             [NotNull] string[] labels)
-        {
-            if (FindPostgresEnum(annotatable, schema, name) is PostgresEnum enumType)
-                return enumType;
-
-            enumType = new PostgresEnum(annotatable, BuildAnnotationName(schema, name));
-            enumType.SetData(labels);
-            return enumType;
-        }
-
-        public static PostgresEnum GetOrAddPostgresEnum(
-            [NotNull] IMutableAnnotatable annotatable,
-            [NotNull] string name,
-            [NotNull] string[] labels)
-            => GetOrAddPostgresEnum(annotatable, null, name, labels);
-
-        public static PostgresEnum FindPostgresEnum(
-            [NotNull] IAnnotatable annotatable,
-            [CanBeNull] string schema,
-            [NotNull] string name)
-        {
-            Check.NotNull(annotatable, nameof(annotatable));
-            Check.NotEmpty(name, nameof(name));
-
-            var annotationName = BuildAnnotationName(schema, name);
-
-            return annotatable[annotationName] == null ? null : new PostgresEnum(annotatable, annotationName);
-        }
-
-        static string BuildAnnotationName(string schema, string name)
-            => NpgsqlAnnotationNames.EnumPrefix + (schema == null ? name : schema + '.' + name);
-
-        public static IEnumerable<PostgresEnum> GetPostgresEnums([NotNull] IAnnotatable annotatable)
         {
             Check.NotNull(annotatable, nameof(annotatable));
+            Check.NullButNotEmpty(schema, nameof(schema));
```

Comment:
I'd prefer reverting most (all?) of the changes here to `GetOrAddPostgresEnum()` and bring back `FindPostgresEnum()`. Even if the latter isn't currently used in our code base, it's a basic metadata lookup tool, which is once again present in EF Core's sequence implementation (and other similar metadata objects).

Reply:
The API changes mattered more for earlier iterations of the PR, when some of the logic was still fuzzy and incomplete. But at this point, I think we have the semantics clear enough that the public API changes don't have much value anymore.

I'll revert most of this now.

Reply:
Makes sense...